aws/iam.md

```diff
@@ -0,0 +1,69 @@
+## 2021-08-10
+
+Not a new gotcha, but a reminder on how to assume role:
+
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": "sts:AssumeRole",
+            "Resource": [
+                "arn:aws:iam::<account_no>:role/Role",
+                "arn:aws:iam::<account_no>:role/Role2"
+            ]
+        }
+    ]
+}
+```
+
+With above inline policy for a user, the user can:
+assume `Role` or `Role2` on their local with aws cli using
+
+```bash
+aws sts assume-role
+    --role-arn "arn:of:role"
+    --role-session-name "name"
+    --profile "profile_name_of_user"
+```
+**:exclamation: Never use default profile for your aws** Instead use `aws configure` to configure a profile that's not default
```

Comment:
Why is this so?

Reply:
If you only work for one place and your user role doesn't have much access without assuming role, it should be fine. Once you have different roles, access keys, and permission, sometimes you will forget and accidentally kill shit when you do `terraform apply -auto-approve`

Reply:
right. for us we generally just use `default` because witout assuming role, its deny all

Reply:
but agree that we should not rely on `default`

Reply:
> right. for us we generally just use `default` because without assuming role, its deny all

I see, so I see that each user does not have any permission other than `sts:AssumeRole` which is the best practice.